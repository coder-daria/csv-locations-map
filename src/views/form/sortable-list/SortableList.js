import React, { memo } from 'react';
import { SortableContainer } from 'react-sortable-hoc';

import { SortableItem } from './sortable-item';

import { List } from './SortableList.styles';

const SortableList = SortableContainer(({ items, isSorting }) => (
  <List>
    {items.map((value, index) => (
      <SortableItem
        key={index}
        index={index} 
        name={value}
        value={index}
        isSorting={isSorting}
      />
    ))}
  </List>
));

export default memo(SortableList);
