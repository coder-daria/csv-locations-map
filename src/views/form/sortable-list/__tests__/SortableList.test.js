import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import SortableList from '../SortableList';

const setup = () => {
  const props = {
    items: ['state', 'city', 'address', 'zip', 'coategory'],
    isSorting: false,
  };

  return shallow(<SortableList {...props} />, { disableLifecycleMethods: true });
};

describe('<SortableList />', () => {
  it('should match the sanpshot', () => {
    const wrapper = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
