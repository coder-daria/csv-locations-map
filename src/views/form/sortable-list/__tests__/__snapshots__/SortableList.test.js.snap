// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortableList /> should match the sanpshot 1`] = `
<Component
  isSorting={false}
  items={
    Array [
      "state",
      "city",
      "address",
      "zip",
      "coategory",
    ]
  }
/>
`;
